File: cpp/225_ImplementStackusingQueues.cpp
class Stack {
public:
    // Push element x onto stack.
    void push(int x) {
        inQueue.push(x);
        for(int i = 0; i < inQueue.size()-1; ++i){
            inQueue.push(inQueue.front());
            inQueue.pop();
        }
    }

    // Removes the element on top of the stack.
    void pop() {
        inQueue.pop();
    }

    // Get the top element.
    int top() {
        return inQueue.front();
    }

    // Return whether the stack is empty.
    bool empty() {
        return inQueue.empty();
    }
private:
    queue<int> inQueue;
};

File: js/225_ImplementStackusingQueues.js
/**
 * @constructor
 */
var Stack = function() {
    this.queue = [];
};

/**
 * @param {number} x
 * @returns {void}
 */
Stack.prototype.push = function(x) {
    this.queue.push(x);
    for(var i = 0 ; i < this.queue.length -1; ++i){
        this.queue.push(this.queue.shift());
    }
};

/**
 * @returns {void}
 */
Stack.prototype.pop = function() {
    this.queue.shift();
};

/**
 * @returns {number}
 */
Stack.prototype.top = function() {
    return this.queue[0];
};

/**
 * @returns {boolean}
 */
Stack.prototype.empty = function() {
    console.log(this.queue.length);
    return this.queue.length < 1;
};